### PR TITLE
[OpPerf] Implement all linalg ops

### PR DESCRIPTION
**Pull request by connorgoggins**
## Description ##
This PR serves to implement all Linear Algebra operators in OpPerf. To achieve this, I created a file (`benchmark/opperf/nd_operations/linalg_operators.py`) with a function (`run_linalg_operators_benchmarks`) to run all Linear Algebra ops. Within this function, I used calls to `run_performance_test` for the linalg ops with specific requirements on input data values and a call to a new function I wrote (`get_all_linalg_operators`) in conjunction with `run_op_benchmarks` for all other linalg ops. I also added a call to `run_linalg_operators_benchmarks` in `opperf.py`.

I also disabled backwards runs on ops that do not support them (`linalg_gelqf`, `linalg_slogdet`, `moments`).

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- A benchmark/opperf/nd_operations/linalg_operators.py
- M benchmark/opperf/opperf.py
- M benchmark/opperf/rules/default_params.py
- M benchmark/opperf/utils/benchmark_utils.py
- M benchmark/opperf/utils/op_registry_utils.py

## Comments ##
Tested on p2.16xl w/ubuntu 16.04 and Mac OS with:
1. Checkout branch and call function `run_linalg_operators_benchmarks` - runs all Linear Algebra ops on relevant data
2. Checkout branch and run `opperf.py` (full run of all ops)

## Performance Results ##
[Group of operator test - all linalg ops (CPU)](https://gist.github.com/connorgoggins/39cfe77a4d33099d8eb9acc311b2d780)
[Full OpPerf test (CPU)](https://gist.github.com/connorgoggins/c07b18d2b4f3123bf756f299df664ff5)

[Group of operator test - all linalg ops (GPU)](https://gist.github.com/connorgoggins/1a57724611c47235e225346d1b9ac1e4)
[Full OpPerf test (GPU)](https://gist.github.com/connorgoggins/f44b4873af0902faf5ff044061382ef9)

@apeforest @access2rohit @ChaiBapchya

### CODE REVIEW

**Comment by connorgoggins** [comment]
@mxnet-label-bot add [pr-awaiting-review]

File: benchmark/opperf/nd_operations/linalg_operators.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""Performance benchmark tests for MXNet NDArray Linear Algebra Operations.
+
+Below 18 Linear Algebra Operators are covered:
+
+['linalg_potri', 'linalg_gemm2', 'linalg_extractdiag', 'linalg_trsm', 'linalg_gelqf', 'linalg_gemm', 'linalg_sumlogdiag',
+'linalg_potrf', 'linalg_makediag', 'linalg_syrk', 'linalg_maketrian', 'linalg_trmm', 'linalg_extracttrian', 'lingalg_makediag',
```
Comment (ChaiBapchya) [inline benchmark/opperf/nd_operations/linalg_operators.py]: looks like lingalg_makediag has been added twice
Also please rebase

Comment (connorgoggins) [reply benchmark/opperf/nd_operations/linalg_operators.py]: Good catch - done!

**Comment by access2rohit** [comment]
@mxnet-label-bot add [pr-awaiting-merge]

**Comment by access2rohit** [comment]
@apeforest can you also review ?

**Pull request merged**